execution_engine_testing/tests/src/test/system_contracts/standard_payment.rs

```diff
@@ -535,15 +538,25 @@ fn independent_standard_payments_should_not_write_the_same_keys() {
     let transforms_from_genesis = &transforms[1];
     let transforms_from_account_1 = &transforms[2];
 
-    // confirm the two deploys have no overlapping writes
+    // Retrieve the payment purse.
+    let payment_purse = builder
+        .get_handle_payment_contract()
+        .named_keys()
+        .get(handle_payment::PAYMENT_PURSE_KEY)
+        .unwrap()
+        .into_uref()
+        .unwrap();
+
+    // Confirm the two deploys have no overlapping writes except for the payment purse balance.
     let common_write_keys = transforms_from_genesis.keys().filter(|k| {
-        matches!(
-            (
-                transforms_from_genesis.get(k),
-                transforms_from_account_1.get(k),
-            ),
-            (Some(Transform::Write(_)), Some(Transform::Write(_)))
-        )
+        *k != &Key::Balance(payment_purse.addr())
+            && matches!(
+                (
+                    transforms_from_genesis.get(k),
+                    transforms_from_account_1.get(k),
+                ),
+                (Some(Transform::Write(_)), Some(Transform::Write(_)))
+            )
```

Comment:
Wouldn't it be clearer if we took an intersection of both sets (`transforms_from_genesis.keys()` and `transforms_from_account_1.keys()`) and verified that it has a single element that is a write to payment purse balance?

Reply:
Yes, but the keys in a `HashMap` are not necessarily sorted, so taking that intersection would require making a new collection (collect into a `HashSet`) or sorting them (collect into a `Vec`), and at that point I don't know which is clearer.

Reply:
To me, definitely `x.keys().into::<HashSet<_>>() && y.keys.into()::<HashSet<_>>()`.

Reply:
It turned out not to be so simple, as we must filter keys that correspond to `Write` transforms.
Overall I think this isn't worth pursuing, as it would make it more complicated, but I'm open to PRs.